external-crates/move/crates/move-transactional-test-runner/src/framework.rs

```diff
@@ -240,42 +235,43 @@ pub trait MoveTestAdapter<'a>: Sized + Send {
                 extra_args,
             ) => {
                 let syntax = syntax.unwrap_or_else(|| self.default_syntax());
-                let data = match data {
-                    Some(f) => f,
-                    None => panic!(
-                        "Expected a script text block following 'run' starting on lines {}-{}",
-                        start_line, command_lines_stop
-                    ),
-                };
-                let state = self.compiled_state();
-                let (script, warning_opt) = match syntax {
-                    SyntaxChoice::Source => {
-                        let (mut units, warning_opt) = compile_source_units(state, data.path())?;
-                        let len = units.len();
-                        if len != 1 {
-                            panic!("Invalid input. Expected 1 compiled unit but got {}", len)
-                        }
-                        let unit = units.pop().unwrap();
-                        match unit {
-                        AnnotatedCompiledUnit::Script(annot_script) => (annot_script.named_script.script, warning_opt),
-                        AnnotatedCompiledUnit::Module(_) => panic!(
-                            "Expected a script text block, not a module, following 'run' starting on lines {}-{}",
-                            start_line, command_lines_stop
-                        ),
-                    }
-                    }
-                    SyntaxChoice::IR => (compile_ir_script(state, data.path())?, None),
-                };
-                let args = self.compiled_state().resolve_args(args)?;
+                let empty_publish_args = <Self::ExtraPublishArgs as Default>::default();
+                let (warnings_opt, output, data, modules) = compile_any(
+                    self,
+                    "publish",
+                    syntax,
+                    name,
+                    number,
+                    start_line,
+                    command_lines_stop,
+                    stop_line,
+                    data,
+                    |adapter, modules| {
+                        adapter.publish_modules(modules, gas_budget, empty_publish_args)
+                    },
+                )
+                .await?;
+                let (module_id, name) = single_entry_function(&modules).unwrap_or_else(|err| {
+                    panic!("{} on lines {}-{}", err, start_line, command_lines_stop)
```

Comment:
Would it be possible to add a note to explain `run` expects on failure?

Reply:
The single_entry_function error gives you that. Let me know if you want anything more and I can add it in another PR :)

Reply:
Er, I meant explicitly explaining that we "Expected exactly one function or one entry function" _because_ we're in **run**.